core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -210,11 +210,6 @@ public boolean isLoadBalancerCreationConfigured(Stack stack, DetailedEnvironment
      * Adds availability sets to instance groups associated with Knox and Oozie target groups.
      * This method updates the JSON parameters associated with certain instance groups.
      *
-     * It's only necessary to add availability sets to the Azure deployment while we
```

Comment:
The more I think about it, using AS is the best approach. That way there is no special code written for node repair. This gives a common baseline across AWS, Azure, and GCP. In each of the cloud providers, we attach nodes to a group and the group to an LB. That is why I felt that the comment should be removed.

Reply:
Hm, this is only running for Azure:
```java
        if (stack.getCloudPlatform().equalsIgnoreCase(CloudPlatform.AZURE.toString())) {
            configureLoadBalancerAvailabilitySets(stack.getName(), loadBalancers);
        }
```

and using the language of `Availability Sets` is a problem since AWS calls them something different (not sure what GCP uses.)

I agree that it'd be nice to use AS for node repair, but I don't know where we stand on that in CB as a whole. Definitely a question for the CB folks.

Reply:
In AWS it is called auto-scaling groups. In GCP it is called instance groups. In both of them, they serve the same purpose to the load-balancer when it comes to node repair.

That is why there is no separate code in CB for Azure node repair.